cmd/abci-cli/abci-cli.go

```diff
@@ -196,6 +202,45 @@ func persistentArgs(line []byte) []string {
 
 //--------------------------------------------------------------------------------
 
+func cmdTest(app *cli.App, c *cli.Context) error {
+	fmt.Println("Running tests")
+
+	var err error
+
+	err = servertest.InitChain(client)
+	fmt.Println("")
+	err = servertest.SetOption(client, "serial", "on")
+	fmt.Println("")
+	err = servertest.Commit(client, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte("abc"), types.CodeType_BadNonce, nil)
+	fmt.Println("")
+	err = servertest.Commit(client, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.Commit(client, []byte{0, 0, 0, 0, 0, 0, 0, 1})
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00}, types.CodeType_BadNonce, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x01}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00, 0x02}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00, 0x03}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00, 0x00, 0x04}, types.CodeType_OK, nil)
+	fmt.Println("")
+	err = servertest.DeliverTx(client, []byte{0x00, 0x00, 0x06}, types.CodeType_BadNonce, nil)
+	fmt.Println("")
+	err = servertest.Commit(client, []byte{0, 0, 0, 0, 0, 0, 0, 5})
+
+	if err != nil {
```

Comment:
@zramsay: Haven’t noticed your comment (visible now in the outdated diff). It looks like there may be many errors, assigned in many checks and `err` in this line is only the last one (or `nil`), so I’m not sure whether previous assignments have any effect. Actually, if there are some errors, but the last commit test is correct, then it looks like this function will return `nil`… We should therefore probably change each `err = […]` assignment to something like `err = err || […]`.

Reply:
I’ve updated the commit, introducing a helper `or` function.